test/relations.test.js

```diff
@@ -1726,6 +2322,24 @@ describe('relations', function () {
       });
     });
 
+    it('should allow to create belongsTo model in beforeCreate hook with promises', function (done) {
+      var mind;
+      Fear.beforeCreate = function (next) {
+        this.mind.create()
```

Comment:
I think it's still valuable to keep the test case. One thing we have to think about is if we allow hooks to return a `Promise` (similar as how we handle remote methods).